### DISCUSSION TITLE
Locations with different sketches (builder and algebra modes)

### DISCUSSION THREAD

**Discussion by gautaz** [gumyr/build123d#834] [Q&A] [answered]
Hello,

I am trying to use locations (like GridLocations) with different sketches at each location.

With algebra mode, I can think of writing something like:

```python
figures = [
    RegularPolygon(radius=5, side_count=5),
    Circle(radius=5),
    ...
]
Sketch() + [
    loc * figures[index]
    for loc, index in GridLocations(length / 2, width / 2, 2, 2)
]
```

**NOTE**: I am not even sure I can get the `index` somehow.

But for builder mode, I cannot even think of what the syntax would be.

Is there a way to do that at all?

**Comment by jdegenstein** [comment, marked as answer]
Have you checked this example 33 from the intro examples? https://build123d.readthedocs.io/en/latest/introductory_examples.html#python-function-and-for-loop

specifically `enumerate(PolarLocations(a / 2, 6)` -- I am not fully following your desired outcome but `zip` may also be useful.

**Reply by gautaz** [reply]
Great! Exactly what I needed.
Thanks @jdegenstein, I tried all the examples a few weeks ago and... I forgot this one :grimacing:.
